12-multiplatform/2_multiplication/2_3_read_answer/wasm/multiply/main.go

```diff
@@ -0,0 +1,60 @@
+package main
+
+import (
+	"fmt"
+	"math/rand/v2"
+	"strconv"
+	"syscall/js"
+)
+
+type multiplication struct {
+	opLeft, opRight int
+}
+
+func main() {
+	m := &multiplication{}
+
+	// Registering Go functions to JavaScript
+	js.Global().Set("generate", js.FuncOf(m.generate))
+	js.Global().Set("validate", js.FuncOf(m.validate))
+
+	// Wait forever
+	<-make(chan struct{})
+}
+
+func (m *multiplication) generate(_ js.Value, _ []js.Value) any {
+	m.opLeft = rand.IntN(11)
+	m.opRight = rand.IntN(11)
+	dom := js.Global().Get("document")
+
+	dom.Call("getElementById", "operand1").Set("innerHTML", m.opLeft)
+	dom.Call("getElementById", "operand2").Set("innerHTML", m.opRight)
+
+	return nil
+}
+
+func (m *multiplication) validate(this js.Value, args []js.Value) any {
+	dom := js.Global().Get("document")
+
+	defer func() {
+		// Reset the contents of the input field after the user clicked Validate
+		dom.Call("getElementById", "providedAnswer").Set("value", "")
```

Comment:
This is an excellent point. Here is what we had in mind:
a/ we should write as little JavaScript as possible. First, because this book is about Golang, and second, because we don't really know how to write good JavaScript (well, at least I don't).
b/ we wanted to cover both retrieving the whole document from the go layer, and receiving values passed by the JavaScript layer.
(c/ we wanted to keep everything short)
Overall, your approach 2/, except we wanted to explain what the parameters of the `js.FuncOf` are.

In `generate`, we went for a/ because it made sense. In `validate`, we wanted to showcase `b`, but we still wanted to reset the `number` field when the answer is checked.